autogpt_platform/backend/backend/blocks/ayrshare/post.py

```diff
@@ -0,0 +1,531 @@
+import logging
+from datetime import datetime
+from typing import List, Optional
+
+from pydantic import BaseModel, Field, SecretStr
+
+from backend.blocks.ayrshare._api import (
+    AyrshareClient,
+    PostError,
+    PostResponse,
+    SocialPlatform,
+)
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema, BlockType
+from backend.data.model import SchemaField
+from backend.integrations.credentials_store import IntegrationCredentialsStore
+
+logger = logging.getLogger(__name__)
+
+creads_store = IntegrationCredentialsStore()
+
+
+class RequestOutput(BaseModel):
+    """Base output model for Ayrshare social media posts."""
+
+    status: str = Field(..., description="Status of the post")
+    id: str = Field(..., description="ID of the post")
+    refId: str = Field(..., description="Reference ID of the post")
+    profileTitle: str = Field(..., description="Title of the profile")
+    post: str = Field(..., description="The post text")
+    postIds: Optional[List[dict]] = Field(
+        description="IDs of the posts on each platform"
+    )
+    scheduleDate: Optional[str] = Field(description="Scheduled date of the post")
+    errors: Optional[List[str]] = Field(description="Any errors that occurred")
+
+
+class AyrsharePostBlockBase(Block):
+    """Base class for Ayrshare social media posting blocks."""
+
+    class Input(BlockSchema):
+        """Base input model for Ayrshare social media posts."""
+
+        post: str = SchemaField(
+            description="The post text to be published", default="", advanced=False
+        )
+        media_urls: List[str] = SchemaField(
```

Comment:
low priority but we typically use the list[str] notation, but that's not documented for shit